### PR TITLE
send # of vids from frontend to backend

### PR DESCRIPTION
**Pull request by gracetian6**
- added input form that allows users to enter number of videos desired as variable `numVideos`
- sends `numVideos` from front end to backend by updating as `?num_videos=` in URL
- `numVideos` is a dummy variable in the backend. My next step is to get that working in backend.

image

- EDIT: deleted form POST and changed to `onclick=js function` instead

### CODE REVIEW

File: client/js/genre.js [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+/**
+ * fetches genre count hashmap from /api/youtube and updates html
+ */
 async function displayMusicGenre() {
-  const response = await fetch("/api/youtube");
-  console.log(response.status);
+  const genreBlock = document.getElementById('genres');
+  const numVideos = getNumVids();
+  const response = await fetch(`/api/youtube?num_videos=${numVideos}`);
   if (response.status != 200) {
```
Comment (laptou) [inline client/js/genre.js:26]: check if the response is `401`, not just if it's not `200`; all of the response codes have distinct meanings and most of them don't mean "you are not logged in / authorized"

**Comment by gracetian6** [comment]
- deleted form POST and added js function instead. Gets rid of unnecessary Steps.

Review this PR on https://github.com/googleinterns/step99-2020/pull/28 before this one! (@ibiyemi)

**Comment by gracetian6** [comment]
> LGTM. In the future, consider adding validation to your input so that the user can't enter that they want `onety-one` videos.

@laptou The html tag of my input is set to `type: number` so that words can't be submitted in the input?

File: client/js/genre.js [hunk 2]
```diff
@@ -1,14 +1,32 @@
...
+const DEFAULT_NUM_VIDS = 10;
+
+/**
+ * extracts num vids from input, otherwise returns default
+ * helper fn for displayMusicGenre()
+ * @returns {number} num videos user wants to retrieve
+ */
+function getNumVids() {
+  const numVideos = numVideosInput.value;
+  return numVideos || DEFAULT_NUM_VIDS;
```
Comment (geekster777) [inline client/js/genre.js:17]: nit: just return `numVideosInput.value || DEFAULT_NUM_VIDS`. Creating an intermediary const doesn't improve readability at all so it's not necessary.

**Pull request merged**